src/App.tsx

```diff
@@ -1,56 +1,62 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'debounce';
+import classNames from 'classnames';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { PeopleList } from './PeopleList';
+import { getCheckQuery } from './helper/getCheckQuery';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [people] = useState<Person[]>(peopleFromServer);
+  const [hasOpen, setHasOpen] = useState(false);
+  const [apliedQuery, setApliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const filteredPeople = useMemo(() => {
+    return people.filter((person) => getCheckQuery(person.name, apliedQuery));
+  }, [people, apliedQuery]);
+
+  const appliedQuery = useCallback(debounce(setApliedQuery, 1000), []);
+
+  const handleChangeQuery = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(e.target.value);
+    appliedQuery(e.target.value);
+  };
 
   return (
     <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
-
-      <div className="dropdown is-active">
-        <div className="dropdown-trigger">
+      {selectedPerson ? (
+        <h1 className="title">
+          {`${selectedPerson?.name} (${selectedPerson?.born} = ${selectedPerson?.died})`}
```

Comment:
You've already checked if selectedPerson exists. Why do you need these `?`

Reply:
my bad